Tests/SourceKitLSPTests/CodeLensTests.swift

```diff
@@ -0,0 +1,115 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LSPTestSupport
+import LanguageServerProtocol
+import SKTestSupport
+import SourceKitLSP
+import XCTest
+
+final class CodeLensTests: XCTestCase {
+  // MARK: - Helpers
+
+  func performCodeLensRequest(text: String, range: Range<Position>? = nil) async throws -> [CodeLens]? {
+    let testClient = try await TestSourceKitLSPClient()
+    let uri = DocumentURI(for: .swift)
+
+    testClient.openDocument(text, uri: uri)
+
+    let request = CodeLensRequest(textDocument: TextDocumentIdentifier(uri))
+    return try await testClient.send(request)
+  }
+
+  // MARK: - Tests
+
+  func testEmpty() async throws {
+    let text = ""
+    let lenses = try await performCodeLensRequest(text: text)
+    XCTAssertEqual(lenses, [])
+  }
+
+  func testSuccessfulCodeLensRequest() async throws {
+    let packageManifest = """
+      let package = Package(
+        name: "MyExecutable",
+        products: [.executable(name: "MyExecutable")],
+        targets: [.target(name: "MyExecutable")]
+      )
+      """
+
+    let project = try await SwiftPMTestProject(
+      files: [:],
+      manifest: packageManifest
+    )
+    let uri = DocumentURI(for: .swift)
+
+    let positions = project.testClient.openDocument(
+      """
+      1️⃣@main2️⃣
+      struct MyApp {
+        public static func main() {}
+      }
+      """,
+      uri: uri
+    )
```

Comment:
The dedicated way of doing this would be to add the source file to the `SwiftPMTestProject`.

```suggestion
    let project = try await SwiftPMTestProject(
      files: [
        "test.swift": """
        1️⃣@main2️⃣
        struct MyApp {
          public static func main() {}
        }
        """
      ],
      manifest: packageManifest
    )

    let (uri, positions) = project.openDocument("test.swift")
```